### DISCUSSION TITLE
Passing a define value to TFT_eSPI

### DISCUSSION THREAD

**Discussion by griffinmt** [Bodmer/TFT_eSPI#2532] [General]
Would it be possible to pass a value in the following:
TFT_eSPI tft = TFT_eSPI();

For example:
TFT_eSPI tft = TFT_eSPI(myGC9A01);

I would then be able to specify which display driver to load etc from within my own sketch.
Would this require a change to TFT_eSPI.cpp ?

I shoul mention that I tried the tft_setup.h in my sketch, but compile gives me this failure:
error: 'TFT_WIDTH' was not declared in this scope

**Comment by Bodmer** [comment]
That is not supported and is unlikely to be added in the future.

**Reply by griffinmt** [reply]
understood.  So I guess I go back to doing it all the old way.
